iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.inl

```diff
@@ -25,50 +25,64 @@ namespace iox
 {
 namespace cxx
 {
+// AXIVION Next Construct AutosarC++19_03-A12.6.1: members are initialized in body before read access
 template <uint64_t Capacity, typename ReturnType, typename... Args>
 template <typename Functor, typename>
 inline storable_function<Capacity, signature<ReturnType, Args...>>::storable_function(const Functor& functor) noexcept
 {
     storeFunctor(functor);
 }
 
+// AXIVION Next Construct AutosarC++19_03-A12.1.5: constructor delegation is not feasible here due
+// to lack of sufficient common initialization
+// AXIVION Next Construct AutosarC++19_03-A12.6.1: members are initialized in body before read access
 template <uint64_t Capacity, typename ReturnType, typename... Args>
 inline storable_function<Capacity, signature<ReturnType, Args...>>::storable_function(
     ReturnType (*function)(Args...)) noexcept
-    : /// @NOLINTJUSTIFICATION we use type erasure in combination with compile time template arguments to restore
+    : // AXIVION Next Construct AutosarC++19_03-M5.2.6: the converted pointer is only used
+      // as its original function pointer type after reconversion (type erasure)
+      // AXIVION Next Construct AutosarC++19_03-A5.2.4: reinterpret_cast is required for type erasure
+      /// @NOLINTJUSTIFICATION we use type erasure in combination with compile time template arguments to restore
       ///                      the correct type whenever the callable is used
       /// @NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
     m_callable(reinterpret_cast<void*>(function))
     , m_invoker(invokeFreeFunction)
 {
     cxx::Expects(function);
 
-    m_operations.copyFunction = copyFreeFunction;
-    m_operations.moveFunction = moveFreeFunction;
+    m_operations.copyFunction = &copyFreeFunction;
+    m_operations.moveFunction = &moveFreeFunction;
     // destroy is not needed for free functions
 }
 
+// AXIVION Next Construct AutosarC++19_03-A12.6.1: members are initialized in body before read access
 template <uint64_t Capacity, typename ReturnType, typename... Args>
 template <typename T, typename>
 inline storable_function<Capacity, signature<ReturnType, Args...>>::storable_function(
     T& object, ReturnType (T::*method)(Args...)) noexcept
 {
-    auto p = &object;
+    const auto p = &object;
+    // AXIVION Next Construct AutosarC++19_03-A7.1.1: type erased functor lambda cannot be const
+    // as it may change by calling its operator() later (hence stored as non-const)
```

Comment:
Will be stored in a non-const object, so this would not work. I am fairly sure it is not easy to fix for all cases at once, again due to type erasure. We cannot store the object in a `const void*` later, and even if we could, this would be lying as invoking anything on it later may change the stored object.